desc/imports.go

```diff
@@ -34,10 +34,6 @@ var (
 // package or when the alternate path is only used from one file (so you don't
 // want the alternate path used when loading every other file), use an
 // ImportResolver instead.
-//
-// Deprecated: the new protobuf runtime (v1.4+) verifies that import paths are
-// correct and that descriptors can be linked during package initialization. So
-// registering alternate paths is no longer useful or necessary.
```

Comment:
I also _**un**_-deprecated these while in here. I was mistaken when I wrote this comment: the protobuf runtime does not verify that import paths are correct. Instead it will happily delay the errors until you need to use reflection. At reflection time, it can provide "placeholder" descriptors for files that are imported using the wrong name. So all of the work-arounds in this file for patching import paths are, in fact, still useful.